code/modules/vending/_vending.dm

```diff
@@ -1145,17 +1160,18 @@ GLOBAL_LIST_EMPTY(vending_machines_to_restock)
 	if(!component_parts || !refill_canister)
 		return FALSE
 
-	var/moved = 0
-	if(panel_open || replacer.works_from_distance)
-		if(replacer.works_from_distance)
-			display_parts(user)
-		for(var/replacer_item in replacer)
-			if(istype(replacer, refill_canister))
-				moved += restock(replacer_item)
-	else
-		display_parts(user)
-	if(moved)
-		to_chat(user, span_notice("[moved] items restocked."))
+	if(!panel_open || replacer.works_from_distance)
+		to_chat(user, display_parts(user))
+
+	if(!panel_open && !replacer.works_from_distance)
+		return FALSE
+
+	var/restocked = 0
+	for(var/replacer_item in replacer)
+		if(istype(replacer_item, refill_canister))
+			restocked += restock(replacer_item)
+	post_restock(user, restocked)
+	if(restocked)
```

Comment:
```suggestion
	if(restocked > 0)
```

this isn't your fault but we should avoid treating integer variables like a boolean in these cases so let's just fix that righ tnow

Reply:
yeah fair